tezos/encoding-derive/src/make.rs

```diff
@@ -0,0 +1,689 @@
+                    param,
+                    span: meta.span(),
+                })
+            } else {
+                Err(error_spanned(meta, "Parameter is required"))
+            }
+        })
+        .transpose()
+}
+
+/// Gets attribute named `name` checking that it does not have any parameters.
+fn get_attribute_no_param(
+    meta: &mut Vec<syn::Meta>,
+    name: &symbol::Symbol,
+) -> Result<Option<AttrWithParam<()>>> {
+    get_attribute(meta, name)
+        .map(|meta| {
+            if let syn::Meta::Path(_) = meta {
+                Ok(AttrWithParam {
+                    param: (),
+                    span: meta.span(),
+                })
+            } else {
+                Err(error_spanned(meta, "Parameter is unexpected"))
+            }
+        })
+        .transpose()
+}
+
+fn make_enum_encoding<'a>(
+    data: &'a syn::DataEnum,
+    name: &'a syn::Ident,
+    meta: &mut Vec<syn::Meta>,
+) -> Result<EnumEncoding<'a>> {
+    let tag_type = get_attribute_value_parsed(meta, &symbol::TAGS)?
+        .unwrap_or_else(|| syn::Ident::new("u8", data.enum_token.span()));
+    let tags = make_tags(&data.variants)?;
+    Ok(EnumEncoding {
+        name,
+        tag_type,
+        tags,
+    })
+}
+
+fn make_tags<'a>(variants: impl IntoIterator<Item = &'a syn::Variant>) -> Result<Vec<Tag<'a>>> {
+    let mut default_id = 0;
+    let mut tags = Vec::new();
+    for variant in variants {
+        let meta = &mut get_encoding_meta(&variant.attrs)?;
+        let tag = make_tag(variant, meta, default_id)?;
+        default_id += 1;
+        tags.push(tag);
+    }
+    Ok(tags)
+}
+
+fn make_tag<'a>(
+    variant: &'a syn::Variant,
+    meta: &mut Vec<syn::Meta>,
+    default_id: u16,
+) -> Result<Tag<'a>> {
+    let id = get_attribute_value(meta, &symbol::TAG)?
+        .map(|lit| {
+            if let syn::Lit::Int(int_lit) = lit {
+                Ok(int_lit)
+            } else {
+                Err(error_spanned(lit, "Integer literal expected"))
+            }
+        })
+        .unwrap_or_else(|| Ok(syn::LitInt::new(&default_id.to_string(), variant.span())))?;
+    let name = &variant.ident;
+    let encoding = match &variant.fields {
+        syn::Fields::Named(_) => {
+            return Err(error_spanned(
+                variant,
+                "Named fields are not supported for enums",
+            ))
+        }
+        syn::Fields::Unnamed(fields) if fields.unnamed.len() == 1 => {
+            let ty = &fields.unnamed.first().unwrap().ty;
+            match ty {
+                syn::Type::Path(type_path) => Encoding::Path(&type_path.path),
+                _ => return Err(error_spanned(ty, "Unsupported type for enum variant")),
+            }
+        }
+        syn::Fields::Unnamed(fields) => {
+            return Err(error_spanned(fields, "Only single field is supported"))
+        }
+        syn::Fields::Unit => Encoding::Unit,
+    };
+    Ok(Tag { id, name, encoding })
+}
+
+/*
```

Comment:
@akoptelov 
do we need this?
if not please remove it

Reply:
Removed.